api/routes/users/get.js

```diff
@@ -1,37 +1,19 @@
-const logger = require('../../logger')('users route get');
 const { getAllUsers: ga, getUserByID: gu } = require('../../db');
 const can = require('../../middleware').can;
 
 module.exports = (app, { getAllUsers = ga, getUserByID = gu } = {}) => {
-  logger.silly('setting up GET /users route');
-  app.get('/users', can('view-users'), async (req, res) => {
-    logger.silly({ id: req.id, message: 'handling GET /users route' });
-    try {
-      const users = await getAllUsers();
-      logger.silly({ id: req.id, message: 'sending users' });
-      res.send(users);
-    } catch (e) {
-      logger.error({ id: req.id, message: e });
-      res.status(500).end();
-    }
+  app.get('/users', can('view-users'), async (req, res, next) => {
+    await getAllUsers()
+      .then(users => res.send(users))
+      .catch(next);
   });
-  logger.silly('setting up GET /users/:id route');
-  app.get('/users/:id', can('view-users'), async (req, res) => {
-    logger.silly({ id: req.id, message: 'handling GET /users/:id route' });
-    logger.silly({ id: req.id, message: 'got a request for a single user', userId: req.params.id });
-    try {
-      const user = await getUserByID(req.params.id);
 
-      if (user) {
-        logger.silly({ id: req.id, message: 'sending user', user });
-        res.send(user);
-      } else {
-        logger.verbose({ id: req.id, message: `no user found [${req.params.id}]`});
-        res.status(404).end();
-      }
-    } catch (e) {
-      logger.error({ id: req.id, message: e });
-      res.status(500).end();
-    }
+  app.get('/users/:id', can('view-users'), async (req, res, next) => {
+    await getUserByID(req.params.id)
```

Comment:
await then

Reply:
Ah, I see. I was sprinkling async/await everywhere when it isn't necessary.

Reply:
you just wanted to make sure it really waited until it was finished :D

Reply:
Unfortunately, the unit tests fail without async/await present